src/utils/string-utils.ts

```diff
@@ -69,12 +69,16 @@ export function formatCurrency(
 export function assetToAmount(asset: string): number {
     try {
         const qty: string = asset.split(' ')[0];
-        return parseFloat(qty);
+        return parseFloat(qty.replace(/,/g, '')); // remove commas
     } catch (error) {
         return 0;
     }
 }
 
+export function formatNumberWithCommas(num: number): string {
+    return num.toString().replace(/\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g, ',');
+}
```

Comment:
I learned with @EJSG that you can delegate this number-formating job to the javascript built-in solution Intl.NumberFormat like this:

```suggestion
export function formatNumberWithCommas(num: number): string {
  return new Intl.NumberFormat('en-US', {
    minimumFractionDigits: 0,
    maximumFractionDigits: 20,
  }).format(num);
}
```